### DISCUSSION TITLE
DB_PREFIX is not recognized in migration dropColumn and  dropForeign

### DISCUSSION THREAD

**Discussion by yespbs** [laravel/framework#33081] [General] [answered]
- Laravel Version: all. tested upto 7.13
- PHP Version: all. tested upto 7.4
- Database Driver & Version: mysql all, tested upto 8

### Description:
DB_PREFIX is not recognized in migration dropColumn and  dropForeign. As prefix creates foreign keys with prefix itself, while removing, it fails to associate the same and halts. To by pass, each case table name must be wrapped with env('DB_PREFIX')

### Steps To Reproduce:
1. Add something in .env DB_PREFIX=sgd_
2. Add a migration with foreign key
3. Try to migrate:rollback

**Comment by armancodes** [comment]
I tried to create the tables using the `DB_PREFIX=foo` in the `.env` file but it didn't work.

Here's the reason:
As I can see in the `config/database.php` file database connections are not using the `env('DB_PREFIX')` instead they used an empty string:
![image](https://user-images.githubusercontent.com/22839699/83678140-7dd4fb00-a5f2-11ea-98bf-1d3c231e7bfd.png)
![image](https://user-images.githubusercontent.com/22839699/83678192-97764280-a5f2-11ea-98e3-716b40f33303.png)

You can just change the value of their `prefix` to something else to see if this works. For me, it worked as expected.

**Comment by yespbs** [comment]
@armancodes 
By default prefix is not enabled, you can enable by adding the env

'prefix' => env('DB_PREFIX', ''), 

which works, all normal create, drop works except for foreign keys added/dropped in a migration. 

This issue is related to dropColumn  and  dropForeign.

If you use dropColumn  with array wrap like dropColumn (['column_name']) it drops  column as well foreign keys associated unless you use DB_PREFIX

**Comment by armancodes** [comment, marked as answer]
Can you share the code and the error?

**Reply by yespbs** [reply]
some how old code was causing trouble but fresh install solved that
please close

Thanks for tip

**Comment by driesvints** [comment]
Moving to discussions.
